### PR TITLE
Issues indicated by FindBugs

### PR DESCRIPTION
**Pull request by BenBE**
This patchset contains some issues raised by FindBugs.

These patches also clean up the code and make it more strict regarding generic type arguments (previously left out).

There are also 4 resource leaks and one infinite loop bug addressed.

### CODE REVIEW

**Comment by BenBE** [comment]
NB: The compile issue results from your current master.

File: TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/ClientHelloDtlsMessage.java [hunk 1]
```diff
@@ -22,10 +22,10 @@
 public class ClientHelloDtlsMessage extends ClientHelloMessage {
 
     @ModifiableVariableProperty(type = ModifiableVariableProperty.Type.COOKIE)
-    ModifiableByteArray cookie;
+    ModifiableByteArray cookie = null;
```
Comment (jurajsomorovsky) [inline TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/ClientHelloDtlsMessage.java]: What is the difference here?
Instance variables are by default initialize with `null`, correct?

File: TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/HelloVerifyRequestHandler.java [hunk 2]
```diff
@@ -23,12 +23,13 @@
  * @author Florian Pfützenreuter <florian.pfuetzenreuter@rub.de>
  * @param <HandshakeMessage>
  */
-public class HelloVerifyRequestHandler<HandshakeMessage extends HelloVerifyRequestMessage> extends
```
Comment (jurajsomorovsky) [inline TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/HelloVerifyRequestHandler.java]: To be honest, I do not understand these changes. What do they bring?

**Comment by jurajsomorovsky** [comment]
Thank you for your pull request. I will take them after the above questions are cleared.

File: TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/ClientHelloDtlsMessage.java (hunk 1 above)
Comment (BenBE) [inline TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/ClientHelloDtlsMessage.java]: FindBugs remarked an uninitialized read on the "UnsafeGet" operation later in the constructor code. Even if implicitly initialized to null I think it's cleaner to write this out here. Handling within the other code is not affected as far as I see.

File: TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/HelloVerifyRequestHandler.java (hunk 2 above)
Comment (BenBE) [inline TLS/src/main/java/de/rub/nds/tlsattacker/dtls/protocol/handshake/HelloVerifyRequestHandler.java]: The Parameter for this Generic Type argument was renamed as there is a type HandshakeMessage which is hidden through this parameter of the same name. Thus it wasn't really clear which type (the parameterized one or HandshakeMessage) was ment when HandshakeMessage was written in the code. Renaming the type argument resolves this disambiguation.

The change in the constructor are due to the field "correctProtocolMessageClass" being of Class<? extends Message>; and thus needing typecast to take in the Class<Object> returned by HelloVerifyRequestMessage.class

**Comment by BenBE** [comment]
Did a rebase onto current master in hopes of the compile/travis issues being fixed. Otherwise unchanged.

**Pull request merged**